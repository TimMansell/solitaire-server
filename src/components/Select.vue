<template>
  <span class="select">
    <select :value="value" @change="$emit('select', $event.target.value)">
      <option v-for="(item, index) in items" :key="index">{{ item }}</option>
    </select>
  </span>
</template>

<script>
export default {
  name: 'Select',
  props: {
    value: {
      type: [String, Number],
      default: '',
    },
    items: {
      type: Array,
      default: () => [],
    },
  },
};
</script>

<style scoped lang="scss">
.select {
  display: inline-block;
  margin-bottom: var(--mg-md);
}
</style>
