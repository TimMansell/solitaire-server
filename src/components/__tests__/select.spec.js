import { shallowMount } from '@vue/test-utils';
import Select from '@/components/Select.vue';

describe('Select.vue', () => {
  it('matches snapshot - number', () => {
    const wrapper = shallowMount(Select, {
      propsData: {
        label: 'Label 1',
        value: 1,
        items: [1, 2, 3],
      },
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot - string', () => {
    const wrapper = shallowMount(Select, {
      propsData: {
        label: 'Label 1',
        value: 'value 2',
        items: ['value 1', 'value 2', 'value 3'],
      },
    });

    expect(wrapper).toMatchSnapshot();
  });
});
