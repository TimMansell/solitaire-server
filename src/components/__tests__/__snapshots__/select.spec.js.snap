// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select.vue matches snapshot - number 1`] = `
<div class="select"><label for="Label-1" class="select__label">Label 1</label> <select id="Label-1" data-test="select" class="select__dropdown">
    <option>
      1
    </option>
    <option>
      2
    </option>
    <option>
      3
    </option>
  </select></div>
`;

exports[`Select.vue matches snapshot - string 1`] = `
<div class="select"><label for="Label-1" class="select__label">Label 1</label> <select id="Label-1" data-test="select" class="select__dropdown">
    <option>
      value 1
    </option>
    <option>
      value 2
    </option>
    <option>
      value 3
    </option>
  </select></div>
`;
